packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,519 @@
+  async getAccountFromProvider(address: Address): Promise<Account> {
+    const accountData = await this.provider.send('eth_getProof', [
+      address.toString(),
+      [],
+      this.blockTag,
+    ])
+
+    const rawData = accountData.accountProof
+
+    await this.trie.fromProof(rawData.map((e: string) => hexToBytes(e)))
+
+    const account = Account.fromAccountData({
+      balance: BigInt(accountData.balance),
+      nonce: BigInt(accountData.nonce),
+      codeHash: toBuffer(accountData.codeHash),
+      storageRoot: toBuffer(accountData.storageHash),
+    })
+    return account
+  }
+
+  /**
+   * Saves an account into state under the provided `address`.
+   * @param address - Address under which to store `account`
+   * @param account - The account to store
+   */
+  async putAccount(address: Address, account: Account): Promise<void> {
+    await this.trie.put(address.buf, account.serialize())
+  }
+
+  /**
+   * Gets the state-root of the Merkle-Patricia trie representation
+   * of the state of this StateManager.
+   * @returns {Buffer} - Returns the state-root of the `StateManager`
+   */
+  async getStateRoot(): Promise<Buffer> {
+    return this.trie.root()
+  }
+
+  /**
+   * Sets the state of the instance to that represented
+   * by the provided `stateRoot`.
+   * @param stateRoot - The state-root to reset the instance to
+   */
+  async setStateRoot(stateRoot: Buffer): Promise<void> {
+    this.trie.root(stateRoot)
+  }
+
+  /**
+   * Checks whether there is a state corresponding to a stateRoot
+   */
+  async hasStateRoot(root: Buffer): Promise<boolean> {
+    const hasRoot = await this.trie.checkRoot(root)
+    return hasRoot
+  }
+
+  /**
+   * Get an EIP-1186 proof
+   * @param address address to get proof of
+   * @param storageSlots storage slots to get proof of
+   * @returns an EIP-1186 formatted proof
+   */
+  async getProof(address: Address, storageSlots: Buffer[] = []): Promise<Proof> {
+    const account = await this.getAccount(address)
+    const accountProof: PrefixedHexString[] = (await this.trie.createProof(address.buf)).map((p) =>
+      bufferToHex(p)
+    )
+    const storageProof: StorageProof[] = []
+    const storageTrie = await this._getStorageTrie(address)
+
+    for (const storageKey of storageSlots) {
+      const proof = (await storageTrie.createProof(storageKey)).map((p) => bufferToHex(p))
+      let value = bufferToHex(await this.getContractStorage(address, storageKey))
+      if (value === '0x') {
+        value = '0x0'
+      }
+      const proofItem: StorageProof = {
+        key: bufferToHex(storageKey),
+        value,
+        proof,
+      }
+      storageProof.push(proofItem)
+    }
+
+    const returnValue: Proof = {
+      address: address.toString(),
+      balance: bigIntToHex(account.balance),
+      codeHash: bufferToHex(account.codeHash),
+      nonce: bigIntToHex(account.nonce),
+      storageHash: bufferToHex(account.storageRoot),
+      accountProof,
+      storageProof: [],
```

Comment:
```suggestion
      storageProof,
```
is it intentional that we're returning an empty array?

Reply:
Nope, just missed this :facepalm: